### PR TITLE
Docs deprecate setup dashboards

### PR DESCRIPTION
**Pull request by simitt**
From 6.4 on loading the dashboards and Kibana index pattern via the APM Server will be deprecated, in favour of loading them via the Kibana UI. 
The reason behind this is that some features will require an existing APM Kibana Index Pattern. If a missing index pattern is detected the customer can load the Index Pattern directly via Kibana, instead of having to load it via the Server. 

This PR aims to 
* print a deprecated message when the dashboards and index patterns are loaded via the server.
* mark Kibana and dashboard sections as deprecated in `apm-server.yml`.
* mark configuring Kibana endpoint and dashboard settings as deprecated in official docs.

implements elastic/apm-server#1135 

update:
forgot to mention, that once we align on the deprecation docs I'll open a PR in beats to update the copied docs properly.

### CODE REVIEW

File: docs/dashboards.asciidoc [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+From APM Server and Kibana 6.4 on you can load dashboards directly via the Kibana UI.
 
-include::./copied-from-beats/dashboards.asciidoc[]
\ No newline at end of file
+NOTE: Loading index pattern and dashboards through the APM Server is deprecated.
+      It is recommended to use the Kibana UI instead.
+      If you still want to load the dashboards via the APM Server,
+      you can do so by following the deprecated instructions:
+
+==== Deprecated Setup
```
Comment (jalvz) [inline docs/dashboards.asciidoc]: I'd remove or reword this heading, feels a bit too aggressive

**Comment by roncohen** [comment]
cool @simitt. I guess the conditional `deprecate_dashboard_loading` is there because you intend to submit the changes to files in `copied-from-beats` back to beats?

**Comment by jalvz** [comment]
that makes sense, thanks 👍

File: docs/copied-from-beats/shared-template-load.asciidoc [hunk 2]
```diff
@@ -94,18 +94,37 @@ that you specify should include the root name of the index plus version and
...
 this option to overwrite the index name defined in the dashboards and index
 pattern.
 
+endif::[]
+
+ifdef::deprecate_dashboard_loading[] 
+
+["source","sh",subs="attributes,callouts"]
+-----
+output.elasticsearch.index: "customname-%{[beat.version]}-%{+yyyy.MM.dd}"
```
[the remark this answers is not in the document]
Comment (simitt) [reply docs/copied-from-beats/shared-template-load.asciidoc]: good question - actually it does render correctly, I only copy & pasted it from above and removed the deprecated part. 
screen shot 2018-07-17 at 15 56 15

**Comment by simitt** [comment]
We just encountered that for OSS users it is not possible to change the index-pattern in the Kibana config with the current implementation. This means that OSS users would still need to use the current handling if they don't want to use the default index names. 
Thus, we should not mark the current setup as deprecated, but only mention a new option. I'll create a new PR linking to the updated Kibana docs once they are up and everything is sorted out. (cc @sqren and @makwarth).

**Pull request merged**